src/random.cpp

```diff
@@ -67,7 +67,7 @@ static inline int64_t GetPerformanceCounter() noexcept
 #endif
 }
 
-#ifdef HAVE_GETCPUID
+#if defined(HAVE_GETCPUID) && !defined(_MSC_VER)
```

Comment:
If `GetCPUID` is now implemented for MSVC, why is this skipped?

Reply:
> If `GetCPUID` is now implemented for MSVC

It is implemented in the first commit.

> why is this skipped?

MSVC does not support inline assembly on x64: https://github.com/bitcoin/bitcoin/blob/593979494042e77b5271d42b8f6e7b514e0007d8/src/random.cpp#L129-L136

Reply:
> It is implemented in the first commit.

I'm aware. That's why I said "is now implemented".

Reply:
1. This `#if ... #else ... #endif` [block](https://github.com/hebasto/bitcoin/blob/915bdbdf8c87e20426161bbf393d1036659813a2/src/random.cpp#L69-L185) guards two static functions--`GetRdRand` and `GetRdSeed`--which are not implemented for Windows. Not adding `&& !defined(_MSC_VER)` ends with an error:
```
C:\Users\hebasto\bitcoin\src\random.cpp(137,1): fatal  error C1189: #error:  "RdRand is only supported on x86 and x86_64" [C:\Users\hebasto\bitcoin\build_msvc\li
bbitcoin_util\libbitcoin_util.vcxproj]
```

2. > And I don't understand the comment about MSVC and assembly.

Windows implementation for the `GetRdRand` and `GetRdSeed` was not added because "MSVC does not support inline assembly on x64".

3. > Was `GetCPUID` not compiling already?

It was. On Windows, it is used in the `SHA256AutoDetect` and `AddCPUID` functions.

Reply:
Hmm, ok, we are talking around each-other. I'll ask some detailed questions in code and see if we can clear it up.